File: translation_output/w_test/mistral/codenet/java/python/p00379.py
<s> [INST] Translate the following java code to python and enclose your solution inside ```python```.
A sample test case is provided below:

Test input:
16 2 1000

Expected output:
3


```
import java.util.Scanner;

class p00379{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int a = sc.nextInt();
		int n = sc.nextInt();
		int m = sc.nextInt();
		int max = (int) Math.pow(m, (double) 1 / n);
		
		int ans = 0;
		for(int i = 1; i <= max; i++) {
			int x = (int) Math.pow(i, n);
			String str = String.valueOf(x);
			int y = 0;
			for(int j = 0; j < str.length(); j++) {
				y += Integer.parseInt(str.substring(j, j + 1));
			}
			
			if(y + a == i) {
				ans++;
			}
		}
		
		System.out.println(ans);
	}
}

```
[/INST]
```python
import math

def p00379():
    a = int(input())
    n = int(input())
    m = int(input())
    max = int(math.pow(m, 1 / n))
    
    ans = 0
    for i in range(1, max+1):
        x = int(math.pow(i, n))
        str = str(x)
        y = 0
        for j in range(len(str)):
            y += int(str[j])
        if y + a == i:
            ans += 1
    
    print(ans)

p00379()
```</s>

File: translation_output/w_test/mistral/codenet/python/java/p00379.java
<s> [INST] Translate the following python code to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
16 2 1000

Expected output:
3


```
a, n, m = map(int, input().split())
ans = 0
for y in range(1, 50):
    x = (y + a)**n
    if 1 <= x <= m and y == sum(map(int, str(x))):
        ans += 1
print(ans)

```
[/INST]
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a, n, m;
        a = sc.nextInt();
        n = sc.nextInt();
        m = sc.nextInt();
        int ans = 0;
        for (int y = 1; y < 50; y++) {
            int x = (y + a) * Math.pow(n, y);
            if (x <= m && y == Integer.parseInt(String.valueOf(x))) {
                ans++;
            }
        }
        System.out.println(ans);
    }
}
```</s>
